### PR TITLE
KOGITO-9088: Add VSCode extension tests for Serverless Workflow YAML files

### PR DESCRIPTION
**Pull request by LuboTerifaj**
…files

https://issues.redhat.com/browse/KOGITO-9088

### CODE REVIEW

File: packages/serverless-workflow-vscode-extension/it-tests/serverless-workflow-editor-extension-basic-operations.test.ts [hunk 1]
```diff
@@ -107,24 +107,84 @@ describe("Serverless workflow editor - Basic operations tests", () => {
...
+    expect((await swfEditor.getAllNodeIds()).length).equal(6);
+
+    let textEditor = await swfTextEditor.getSwfTextEditor();
+
+    const greetInGermanStateString =
+      "- name: GreetInGerman\n" +
+      "  type: inject\n" +
+      "  data:\n" +
+      "    greeting: 'Hallo vom JSON-Workflow, '\n" +
+      "  transition: GreetPerson";
```
Comment (zdrapela) [inline packages/serverless-workflow-vscode-extension/it-tests/serverless-workflow-editor-extension-basic-operations.test.ts]:
```suggestion
      "  - name: GreetInGerman\n" +
      "    type: inject\n" +
      "    data:\n" +
      "      greeting: 'Hallo vom JSON-Workflow, '\n" +
      "    transition: GreetPerson";
```

File: packages/serverless-workflow-vscode-extension/it-tests/serverless-workflow-editor-extension-basic-operations.test.ts [hunk 2]
```diff
@@ -107,24 +107,84 @@ describe("Serverless workflow editor - Basic operations tests", () => {
...
+
+    const germanConditionString = '  - condition: ${ .language == "German" }\n' + "    transition: GreetInGerman";
```
Comment (zdrapela) [inline packages/serverless-workflow-vscode-extension/it-tests/serverless-workflow-editor-extension-basic-operations.test.ts]:
```suggestion
    const germanConditionString = '      - condition: ${ .language == "German" }\n' + "        transition: GreetInGerman";
```

File: packages/serverless-workflow-vscode-extension/it-tests/serverless-workflow-editor-extension-basic-operations.test.ts [hunk 3]
```diff
@@ -107,24 +107,84 @@ describe("Serverless workflow editor - Basic operations tests", () => {
...
+
+    await textEditor.typeTextAt(30, 26, "\n");
```
Comment (zdrapela) [inline packages/serverless-workflow-vscode-extension/it-tests/serverless-workflow-editor-extension-basic-operations.test.ts]:
```suggestion
    await textEditor.typeTextAt(30, 28, "\n");
```

File: packages/serverless-workflow-vscode-extension/it-tests/serverless-workflow-editor-extension-basic-operations.test.ts [hunk 4]
```diff
@@ -107,24 +107,84 @@ describe("Serverless workflow editor - Basic operations tests", () => {
...
+    await textEditor.setTextAtLine(31, greetInGermanStateString);
+    await textEditor.typeTextAt(18, 31, "\n");
```
Comment (zdrapela) [inline packages/serverless-workflow-vscode-extension/it-tests/serverless-workflow-editor-extension-basic-operations.test.ts]:
```suggestion
    await textEditor.typeTextAt(18, 35, "\n");
```

File: packages/serverless-workflow-vscode-extension/it-tests/serverless-workflow-editor-extension-diagram-navigation.test.ts [hunk 5]
```diff
@@ -79,4 +79,36 @@ describe("Serverless workflow editor - Diagram navigation tests", () => {
...
+
+    // Select CheckApplication node
+    await swfEditor.selectNode(nodeIds[1]);
+
+    const textEditor = await swfTextEditor.getSwfTextEditor();
+    let lineNumber = (await textEditor.getCoordinates())[0];
+    let columnNumber = (await textEditor.getCoordinates())[1];
+
+    expect(lineNumber).equal(11);
+    expect(columnNumber).equal(3);
```
Comment (zdrapela) [inline packages/serverless-workflow-vscode-extension/it-tests/serverless-workflow-editor-extension-diagram-navigation.test.ts]:
```suggestion
    expect(columnNumber).equal(5);
```

File: packages/serverless-workflow-vscode-extension/it-tests/serverless-workflow-editor-extension-diagram-navigation.test.ts [hunk 6]
```diff
@@ -79,4 +79,36 @@ describe("Serverless workflow editor - Diagram navigation tests", () => {
...
+
+    // Select StartApplication node
+    await swfEditor.selectNode(nodeIds[2]);
+
+    lineNumber = (await textEditor.getCoordinates())[0];
+    columnNumber = (await textEditor.getCoordinates())[1];
+
+    expect(lineNumber).equal(20);
+    expect(columnNumber).equal(3);
```
Comment (zdrapela) [inline packages/serverless-workflow-vscode-extension/it-tests/serverless-workflow-editor-extension-diagram-navigation.test.ts]:
```suggestion
    expect(columnNumber).equal(5);
```

**Review by zdrapela — CHANGES_REQUESTED** [review]
Hi @LuboTerifaj,
I believe that these changes in whitespace and column numbers are needed. Please see the suggestions. :slightly_smiling_face:

File: packages/serverless-workflow-vscode-extension/it-tests/serverless-workflow-editor-extension-basic-operations.test.ts (hunk 3 above)
Comment (LuboTerifaj) [reply packages/serverless-workflow-vscode-extension/it-tests/serverless-workflow-editor-extension-basic-operations.test.ts]: Yes, you are right. The formatter changed the yaml file. I will update it.

File: packages/serverless-workflow-vscode-extension/it-tests/serverless-workflow-editor-extension-basic-operations.test.ts (hunk 4 above)
Comment (LuboTerifaj) [reply packages/serverless-workflow-vscode-extension/it-tests/serverless-workflow-editor-extension-basic-operations.test.ts]: Yes, you are right. The formatter changed the yaml file. I will update it.

File: packages/serverless-workflow-vscode-extension/it-tests/serverless-workflow-editor-extension-diagram-navigation.test.ts (hunk 5 above)
Comment (LuboTerifaj) [reply packages/serverless-workflow-vscode-extension/it-tests/serverless-workflow-editor-extension-diagram-navigation.test.ts]: The formatter changed the yaml file. I will update it.

File: packages/serverless-workflow-vscode-extension/it-tests/serverless-workflow-editor-extension-diagram-navigation.test.ts (hunk 6 above)
Comment (LuboTerifaj) [reply packages/serverless-workflow-vscode-extension/it-tests/serverless-workflow-editor-extension-diagram-navigation.test.ts]: The formatter changed the yaml file. I will update it.

**Review by zdrapela — APPROVED** [review]
Thanks, it looks good now!

**Comment by LuboTerifaj** [comment]
Hello @paulovmr , could you please have a look at this PR? Please squash the commits when merging and keep only the main commit message. Thank you!

**Review by paulovmr — APPROVED** [review]

**Pull request merged**